### PR TITLE
Opencv shell scripts

### PR DESCRIPTION
**Pull request by MicLi01**
installation script for opencv

### CODE REVIEW

File: scripts/lib/install_opencv.sh [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+                          python3-dev python3-tk pylint3 python3-numpy flake8
+  sudo apt-get install -y ant default-jdk
+  sudo apt-get install -y doxygen unzip wget
+
+  # Install Library
+  wget https://github.com/opencv/opencv/archive/${OPENCV_VERSION}.zip
+  unzip ${OPENCV_VERSION}.zip && rm ${OPENCV_VERSION}.zip
+  mv opencv-${OPENCV_VERSION} OpenCV
+
+  if [ $OPENCV_CONTRIB = 'YES' ]; then
```
[the remark this answers is not in the document]
Comment (MicLi01) [reply scripts/lib/install_opencv.sh]: fixed

File: scripts/lib/install_opencv.sh [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    wget https://github.com/opencv/opencv_contrib/archive/${OPENCV_VERSION}.zip
+    unzip ${OPENCV_VERSION}.zip && rm ${OPENCV_VERSION}.zip
+    mv opencv_contrib-${OPENCV_VERSION} opencv_contrib
+    mv opencv_contrib OpenCV
+  fi
+
+  cd OpenCV && mkdir build && cd build
+
+  if [ $OPENCV_CONTRIB = 'NO' ]; then
```
[the remark this answers is not in the document]
Comment (MicLi01) [reply scripts/lib/install_opencv.sh]: fixed

File: scripts/lib/install_opencv.sh [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  cmake -DWITH_QT=ON -DWITH_OPENGL=ON -DFORCE_VTK=ON -DWITH_TBB=ON -DWITH_GDAL=ON \
+        -DWITH_XINE=ON -DENABLE_PRECOMPILED_HEADERS=OFF -DWITH_FFMPEG=OFF ..
+  fi
+
+  if [ $OPENCV_CONTRIB = 'YES' ]; then
```
[the remark this answers is not in the document]
Comment (MicLi01) [reply scripts/lib/install_opencv.sh]: fixed

File: scripts/lib/install_opencv.sh [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        -DWITH_XINE=ON -DENABLE_PRECOMPILED_HEADERS=OFF \
+        -DOPENCV_EXTRA_MODULES_PATH=../opencv_contrib/modules ..
+  fi
+
+  make -j8
+  sudo make install
+  sudo ldconfig
+}
+
+command -v opencv > /dev/null 2>&1 || { install_opencv; }
```
[the remark this answers is not in the document]
Comment (MicLi01) [reply scripts/lib/install_opencv.sh]: removed

File: scripts/lib/install_opencv.sh [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  sudo apt-get install -y libtbb-dev libeigen3-dev
+  sudo apt-get install -y python-dev  python-tk  pylint  python-numpy  \
+                          python3-dev python3-tk pylint3 python3-numpy flake8
+  sudo apt-get install -y ant default-jdk
+  sudo apt-get install -y doxygen unzip wget
+
+  # Install Library
+  wget https://github.com/opencv/opencv/archive/${OPENCV_VERSION}.zip
+  unzip ${OPENCV_VERSION}.zip && rm ${OPENCV_VERSION}.zip
+  mv opencv-${OPENCV_VERSION} OpenCV
```
[the remark this answers is not in the document]
Comment (MicLi01) [reply scripts/lib/install_opencv.sh]: Made a new directory variable called OPENCV_DIRECTORY

File: scripts/lib/install_opencv.sh [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+        -DWITH_XINE=ON -DENABLE_PRECOMPILED_HEADERS=OFF -DWITH_FFMPEG=OFF ..
+  fi
+
+  if [ $OPENCV_CONTRIB = 'YES' ]; then
+  cmake -DWITH_QT=ON -DWITH_OPENGL=ON -DFORCE_VTK=ON -DWITH_TBB=ON -DWITH_GDAL=ON \
+        -DWITH_XINE=ON -DENABLE_PRECOMPILED_HEADERS=OFF \
+        -DOPENCV_EXTRA_MODULES_PATH=../opencv_contrib/modules ..
+  fi
+
+  make -j8
```
[the remark this answers is not in the document]
Comment (MicLi01) [reply scripts/lib/install_opencv.sh]: ok

File: scripts/install [hunk 7]
```diff
@@ -21,6 +23,9 @@ else
     kafka)
       source ./lib/install_kafka.sh
       install_kafka
```
Comment (chc5) [inline scripts/install]: Add ;;

**Comment by chc5** [comment]
`CMake Warning at cmake/OpenCVFindLibsGUI.cmake:18 (find_package):
  By not providing "FindQt5Core.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "Qt5Core", but
  CMake did not find one.

  Could not find a package configuration file provided by "Qt5Core" with any
  of the following names:

    Qt5CoreConfig.cmake
    qt5core-config.cmake

  Add the installation prefix of "Qt5Core" to CMAKE_PREFIX_PATH or set
  "Qt5Core_DIR" to a directory containing one of the above files.  If
  "Qt5Core" provides a separate development package or SDK, be sure it has
  been installed.
Call Stack (most recent call first):
  CMakeLists.txt:771 (include)


CMake Warning at cmake/OpenCVFindLibsGUI.cmake:19 (find_package):
  By not providing "FindQt5Gui.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "Qt5Gui", but
  CMake did not find one.

  Could not find a package configuration file provided by "Qt5Gui" with any
  of the following names:

    Qt5GuiConfig.cmake
    qt5gui-config.cmake

  Add the installation prefix of "Qt5Gui" to CMAKE_PREFIX_PATH or set
  "Qt5Gui_DIR" to a directory containing one of the above files.  If "Qt5Gui"
  provides a separate development package or SDK, be sure it has been
  installed.
Call Stack (most recent call first):
  CMakeLists.txt:771 (include)


CMake Warning at cmake/OpenCVFindLibsGUI.cmake:20 (find_package):
  By not providing "FindQt5Widgets.cmake" in CMAKE_MODULE_PATH this project
  has asked CMake to find a package configuration file provided by
  "Qt5Widgets", but CMake did not find one.

  Could not find a package configuration file provided by "Qt5Widgets" with
  any of the following names:

    Qt5WidgetsConfig.cmake
    qt5widgets-config.cmake

  Add the installation prefix of "Qt5Widgets" to CMAKE_PREFIX_PATH or set
  "Qt5Widgets_DIR" to a directory containing one of the above files.  If
  "Qt5Widgets" provides a separate development package or SDK, be sure it has
  been installed.
Call Stack (most recent call first):
  CMakeLists.txt:771 (include)


CMake Warning at cmake/OpenCVFindLibsGUI.cmake:21 (find_package):
  By not providing "FindQt5Test.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "Qt5Test", but
  CMake did not find one.

  Could not find a package configuration file provided by "Qt5Test" with any
  of the following names:

    Qt5TestConfig.cmake
    qt5test-config.cmake

  Add the installation prefix of "Qt5Test" to CMAKE_PREFIX_PATH or set
  "Qt5Test_DIR" to a directory containing one of the above files.  If
  "Qt5Test" provides a separate development package or SDK, be sure it has
  been installed.
Call Stack (most recent call first):
  CMakeLists.txt:771 (include)


CMake Warning at cmake/OpenCVFindLibsGUI.cmake:22 (find_package):
  By not providing "FindQt5Concurrent.cmake" in CMAKE_MODULE_PATH this
  project has asked CMake to find a package configuration file provided by
  "Qt5Concurrent", but CMake did not find one.

  Could not find a package configuration file provided by "Qt5Concurrent"
  with any of the following names:

    Qt5ConcurrentConfig.cmake
    qt5concurrent-config.cmake

  Add the installation prefix of "Qt5Concurrent" to CMAKE_PREFIX_PATH or set
  "Qt5Concurrent_DIR" to a directory containing one of the above files.  If
  "Qt5Concurrent" provides a separate development package or SDK, be sure it
  has been installed.
Call Stack (most recent call first):
  CMakeLists.txt:771 (include)


CMake Error at /usr/share/cmake-3.10/Modules/FindQt4.cmake:1320 (message):
  Found unsuitable Qt version "" from NOTFOUND, this code requires Qt 4.x
Call Stack (most recent call first):
  cmake/OpenCVFindLibsGUI.cmake:34 (find_package)
  CMakeLists.txt:771 (include)


-- Configuring incomplete, errors occurred!
See also "/home/chc/Desktop/meowl/scripts/OpenCV/build/CMakeFiles/CMakeOutput.log".
See also "/home/chc/Desktop/meowl/scripts/OpenCV/build/CMakeFiles/CMakeError.log".
make: *** No targets specified and no makefile found.  Stop.
make: *** No rule to make target 'install'.  Stop.
`
Is this normal?

**Comment by chc5** [comment]
```
-- Could NOT find PythonLibs (missing: PYTHON_LIBRARIES PYTHON_INCLUDE_DIRS) (Required is exact version "2.7.17")
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: No module named numpy.distutils
-- Found PythonInterp: /usr/bin/python3 (found suitable version "3.6.9", minimum required is "3.2") 
-- Could NOT find PythonLibs (missing: PYTHON_LIBRARIES PYTHON_INCLUDE_DIRS) (Required is exact version "3.6.9")
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'distutils.sysconfig'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'numpy'

numpy needs to be installed

```

File: scripts/install (hunk 7 above)
Comment (MicLi01) [reply scripts/install]: done

**Pull request merged**